src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -144,39 +144,20 @@ public string MaskSecrets(string input)
         {
             if (object.Equals(currentDetection, null))
             {
-                currentDetection =
-                    new Detection(detection.Id,
-                                  detection.Name,
-                                  detection.Start,
-                                  detection.Length,
-                                  detection.Metadata,
-                                  detection.RotationPeriod,
-                                  detection.CrossCompanyCorrelatingId,
-                                  detection.RedactionToken);
-
+                currentDetection = new Detection(detection);
```

Comment:
Is the copying here because Detection is mutable? Related: #169. It would be nice to make it immutable and then we wouldn't need to make extra instances. Not for this PR.

Reply:
um, looking at this, i'm not sure why these copies are necessary. i just reflexively switched to the alternate ctor to minimize vertical space. let's fix this up in #169.

Reply:
Added a comment to #169